File: tests/nodejs/cases/case_111_alias_shadow_cache_02.js
/**
 * Task case 111: alias_shadow_cache_02 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case111AliasShadowCache02(input) {
  const taskName = 'alias_shadow_cache_02';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const alias = payload;
  // ESCAPE: alias of payload is retained (aliasing false-negative stress).
  retainedCache.set('case_111', alias);
  return 'ok';
}

module.exports = {
  case111AliasShadowCache02,
};

File: tests/python/cases/case_111_alias_shadow_cache_02.py
"""
Task case 111: alias_shadow_cache_02 false-positive/false-negative stress pattern.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_111_alias_shadow_cache_02(input_data):
    task_name = 'alias_shadow_cache_02'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'stress',
        'stage': 'evaluation',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    alias = payload
    # ESCAPE: alias of payload is stored in retained cache (aliasing false-negative stress).
    RETAINED_CACHE['case_111'] = alias
    return 'ok'
